rl4j/rl4j-core/src/main/java/org/deeplearning4j/rl4j/util/LegacyMDPWrapper.java

```diff
@@ -32,9 +32,12 @@ public LegacyMDPWrapper(MDP<O, A, AS> wrappedMDP, IHistoryProcessor historyProce
     }
 
     private IHistoryProcessor getHistoryProcessor() {
-        return historyProcessor;
+        if(historyProcessor != null) {
+            return historyProcessor;
+        }
+        
+        return null;
```

Comment:
Maybe we don't need to do this? :) In any case, please make sure that A3CCartpole isn't crashing!

Reply:
A3CCartpole seems to be running without crashing. I tried it several times because with async, it may be a 'timing' problem that is hard to reproduce. But I think we're fine.

On the other hand, I just noticed that A3CCartpole is not learning. Please do not merge, I will investigate what is wrong.

Reply:
A3CCartpole never really learned that much. It's quite hard to tweak, so don't worry about it...

Reply:
Please simplify (again) the getHistoryProcessor() method though.